packages/contracts-bedrock/src/dispute/AnchorStateRegistry.sol

```diff
@@ -180,60 +199,77 @@ contract AnchorStateRegistry is Initializable, ISemver {
-            emit AnchorNotUpdated(game);
-            return;
+        // Game must be beyond the airgap period.
+        if (!isGameAirgapped(_game)) {
+            return false;
         }
 
-        // Update the anchor game.
-        anchorGame = game;
-        emit AnchorUpdated(game);
+        return true;
     }
 
-    /// @notice Sets the anchor state given the game. Can only be triggered by the Guardian
-    ///         address. Unlike tryUpdateAnchorState(), this function does not check if the
-    ///         provided is newer than the existing anchor game. This allows the Guardian to
-    ///         recover from situations in which the current anchor game is invalid.
-    /// @param _game The game to set the anchor state for.
-    function setAnchorState(IFaultDisputeGame _game) external {
-        // Function can only be triggered by the guardian.
-        if (msg.sender != superchainConfig.guardian()) {
-            revert AnchorStateRegistry_Unauthorized();
+    /// @notice Returns whether a game's root claim is valid.
+    /// @param _game The game to check.
+    /// @return Whether the game's root claim is valid.
+    function isGameClaimValid(IDisputeGame _game) public view returns (bool) {
+        // Game must be a proper game.
+        bool properGame = isGameProper(_game);
+        if (!properGame) {
+            return false;
+        }
+
+        // Must be respected.
+        bool respected = isGameRespected(_game);
+        if (!respected) {
+            return false;
         }
 
-        // Check if the game is a proper game.
-        if (!isGameProper(_game)) {
-            revert AnchorStateRegistry_ImproperAnchorGame();
+        // Game must be finalized.
+        bool finalized = isGameFinalized(_game);
+        if (!finalized) {
+            return false;
         }
 
-        // The game must have resolved in favor of the root claim.
+        // Game must be resolved in favor of the defender.
         if (_game.status() != GameStatus.DEFENDER_WINS) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /// @notice Updates the anchor game.
+    /// @param _game New candidate anchor game.
+    function setAnchorState(IDisputeGame _game) public {
+        // Convert game to FaultDisputeGame.
+        // We can't use FaultDisputeGame in the interface because this function is called from the
+        // FaultDisputeGame contract which can't import IFaultDisputeGame by convention. We should
+        // likely introduce a new interface (e.g., StateDisputeGame) that can act as a more useful
+        // version of IDisputeGame in the future.
+        IFaultDisputeGame game = IFaultDisputeGame(address(_game));
+
+        // Check if the candidate game is valid.
+        bool valid = isGameClaimValid(game);
+        if (!valid) {
+            revert AnchorStateRegistry_InvalidAnchorGame();
+        }
+
+        // Must be newer than the current anchor game.
```

Comment:
Add comment: this will be blocked/bricked if `getAnchorRoot` reverts because the anchor game is blacklisted, intentional